phpunit/fonts-api/gutenbergAddRegisteredFontsToThemeJson-test.php

```diff
@@ -0,0 +1,252 @@
+	 *
+	 * @param string $theme    Theme to use.
+	 * @param array  $fonts    Fonts to register.
+	 * @param array  $expected Expected fonts to be added.
+	 */
+	public function test_should_add_non_theme_json_fonts( $theme, $fonts, $expected ) {
+		switch_theme( static::FONTS_THEME );
+
+		// Register the fonts.
+		wp_register_fonts( $fonts );
+
+		$data   = new WP_Theme_JSON_Gutenberg( self::$theme_json_data[ $theme ] );
+		$actual = gutenberg_add_registered_fonts_to_theme_json( $data );
+
+		$this->assertNotSame( $data, $actual, 'New instance of WP_Theme_JSON_Gutenberg should be returned' );
+		$actual_raw_data = $actual->get_raw_data();
+
+		$this->assertArrayHasKey( 'typography', $actual_raw_data['settings'] );
+		$this->assertArrayHasKey( 'fontFamilies', $actual_raw_data['settings']['typography'] );
+		$this->assertArrayHasKey( 'theme', $actual_raw_data['settings']['typography']['fontFamilies'] );
+
+		$this->assertContains(
+			$expected,
+			$actual_raw_data['settings']['typography']['fontFamilies']['theme'],
+			'Fonts should be added after running gutenberg_add_registered_fonts_to_theme_json()'
+		);
+	}
+
+	/**
+	 * Data Provider.
+	 *
+	 * @return array
+	 */
+	public function data_should_add_non_theme_json_fonts() {
+		$lato = array(
+			'Lato' => array(
+				array(
+					'font-family' => 'Lato',
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+					'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular.woff2',
+				),
+				array(
+					'font-family' => 'Lato',
+					'font-style'  => 'italic',
+					'font-weight' => '400',
+					'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular-Italic.woff2',
+				),
+			),
+		);
+
+		$expected_lato = array(
+			'fontFamily' => 'Lato',
+			'name'       => 'Lato',
+			'slug'       => 'lato',
+			'fontFace'   => array(
+				'lato-400-normal' => array(
+					'origin'      => 'gutenberg_wp_fonts_api',
+					'provider'    => 'local',
+					'fontFamily'  => 'Lato',
+					'fontStyle'   => 'normal',
+					'fontWeight'  => '400',
+					'fontDisplay' => 'fallback',
+					'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular.woff2',
+				),
+				'lato-400-italic' => array(
+					'origin'      => 'gutenberg_wp_fonts_api',
+					'provider'    => 'local',
+					'fontFamily'  => 'Lato',
+					'fontStyle'   => 'italic',
+					'fontWeight'  => '400',
+					'fontDisplay' => 'fallback',
+					'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular-Italic.woff2',
+				),
+			),
+		);
+
+		return array(
+			'theme with no fonts defined'              => array(
+				'theme'    => 'block-theme',
+				'fonts'    => $lato,
+				'expected' => $expected_lato,
+			),
+			'theme with fonts: new fonts not in theme' => array(
+				'theme'    => static::FONTS_THEME,
+				'fonts'    => $lato,
+				'expected' => $expected_lato,
+			),
+
+			/*
+			 * @TODO Add these tests fixing https://github.com/WordPress/gutenberg/issues/50047.
```

Comment:
This code doesn't seem to be used at the moment (I understand it will be used at some point in the future). Would it be possible to move it to another (temporary?) branch?
https://github.com/piotrplenik/clean-code-php#remove-dead-code

Reply:
> I think it should be included to gain the advantage of having all expectations defined, even if in this case it's commented out due to a bug.

I understand your point of view.
Personally, I always try to remove dead code, but in this case, I can see how it serves as documentation. Thank you for explaining! Let's leave it.